### PR TITLE
Add version key to setup_remote_docker in configuration reference.

### PR DESCRIPTION
**Pull request by teesloane**
Added missing `version` key to `setup_remote_docker`.

### CODE REVIEW

File: jekyll/_cci2/configuration-reference.md [hunk 1]
```diff
@@ -674,10 +674,25 @@ Key | Required | Type | Description
...
-***Notes***:
+**Notes**:
+
 - A paid account is required to access Docker Layer Caching. Customers on paid plans can request access by [opening a support ticket](https://support.circleci.com/hc/en-us/requests/new). Please include a link to the project on CircleCI) with your request.
 - `setup_remote_docker` is not compatible with the `machine` executor. See [Docker Layer Caching in Machine Executor]({{ site.baseurl }}/2.0/docker-layer-caching/#machine-executor) for information on how to enable DLC with the `machine` executor.
 
+###### **`version`**
+
+If your job requires a specific docker version, you can set it with the
+`version` attribute.
```
Comment (glenjamin) [inline jekyll/_cci2/configuration-reference.md]: ~ could we link off the to version list on the docker page here?

Comment (teesloane) [reply jekyll/_cci2/configuration-reference.md]: Good idea. Done!

File: jekyll/_cci2/configuration-reference.md [hunk 2]
```diff
@@ -674,10 +674,26 @@ Key | Required | Type | Description
 docker_layer_caching | N | boolean | set this to `true` to enable [Docker Layer Caching]({{ site.baseurl }}/2.0/docker-layer-caching/) in the Remote Docker Environment (default: `false`)
```
Comment (michelle-luna) [inline jekyll/_cci2/configuration-reference.md]:
```suggestion
docker_layer_caching | N | boolean | set this to `true` to enable [Docker Layer Caching]({{ site.baseurl }}/2.0/docker-layer-caching/) in the Remote Docker Environment (default: `false`) 
version | N        | String | Version string of Docker you would like to use (default: `17.09.0-ce`). View the list of supported docker versions [here]({{site.baseurl}}/2.0/building-docker-images/#docker-version). **Note**: The version key is not currently supported on CircleCI installed in your private cloud or datacenter. Contact your system administrator for information about the Docker version installed in your remote Docker environment.
{: class="table table-striped"}
```

File: jekyll/_cci2/configuration-reference.md [hunk 3]
```diff
@@ -671,12 +671,16 @@ Creates a remote Docker environment configured to execute Docker commands. See [
 
 Key | Required | Type | Description
 ----|-----------|------|------------
-docker_layer_caching | N | boolean | set this to `true` to enable [Docker Layer Caching]({{ site.baseurl }}/2.0/docker-layer-caching/) in the Remote Docker Environment (default: `false`)
+docker_layer_caching | N | boolean | set this to `true` to enable [Docker Layer Caching]({{ site.baseurl }}/2.0/docker-layer-caching/) in the Remote Docker Environment (default: `false`) 
+version | N        | String | Version string of Docker you would like to use (default: `17.09.0-ce`). View the list of supported docker versions [here]({{site.baseurl}}/2.0/building-docker-images/#docker-version). **Note**: The version key is not currently supported on CircleCI installed in your private cloud or datacenter. Contact your system administrator for information about the Docker version installed in your remote Docker environment.
```
Comment (michelle-luna) [inline jekyll/_cci2/configuration-reference.md]:
```suggestion
version | N        | String | Version string of Docker you would like to use (default: `17.09.0-ce`). View the list of supported docker versions [here]({{site.baseurl}}/2.0/building-docker-images/#docker-version). 
```

**Pull request merged**